helix-core/src/main/java/org/apache/helix/common/caches/CurrentStateCache.java

```diff
@@ -136,8 +136,8 @@ private void refreshCurrentStatesCache(HelixDataAccessor accessor,
     cachedKeys.retainAll(currentStateKeys);
 
     Map<PropertyKey, CurrentState> newStateCache = Collections.unmodifiableMap(
-        refreshProperties(accessor, new ArrayList<>(reloadKeys), new ArrayList<>(cachedKeys),
-            _currentStateCache));
+        refreshProperties(accessor, reloadKeys, new ArrayList<>(cachedKeys),
+            _currentStateCache, reloadKeys));
```

Comment:
This looks strange to me... Let's just have a new reloadedKeyList in addition?

Reply:
This is crucial for fixing the bug. We need reloadKeys to be modified to reflect the change in refreshProperties

Reply:
What you need is a list contains the reloaded key. It does not have to be the original "reloadKeys". The current code put the same object as the input and the output. That means the input will be modified. This just loses the benefit of splitting the input and the output parameters.
Actually, I'm OK with both design: 1. one parameter, or 2. two parameters but sending with different objects. Prefered 2nd one.
However, two parameters but sending the same object looks strange.